src/Tasks/LegalProposition/Config.hs

```diff
@@ -69,7 +69,7 @@ checkAdditionalConfig config@LegalPropositionConfig {syntaxTreeConfig = SynTreeC
     | formulas < illegals + bracketFormulas = reject $ do
         english "The number of formulas cannot be less than the sum of bracket Formulas and illegal ones."
         german "Die Anzahl der Formeln kann nicht niedriger als die Summe von falschen und richtigen Formeln."
-    | let leaves = maxLeavesForNodes maxNodes, (if allowArrowOperators then 4 else 2) ^ (maxNodes - leaves) < formulas
+    | let leaves = maxLeavesForNodes maxNodes, (if arrowOperatorsAllowed treeCfg then 4 else 2) ^ (maxNodes - leaves) < formulas
```

Comment:
Not sure this check is still exact enough now. Previously, `allowArrowOperators` meant that all five binary operators were allowed, and otherwise only two binary opeators were allowed. And this somehow informed the choive between `4` and `2` as the base here. But now `arrowOperatorsAllowed` just checks whether one of the three arrow operators is allowed. It would even be true if overall only one binary operator is allowed at all (with `and` and `or` also disallowed). So the decision for `4` instead of `2` then might be completely wrong.

Put differently, previously the decision here was `if five operators allowed, then take 4 as base for the exponentiation, if two operators allowed, then take 2 as base for the exponentiation`, but not even with just one operator allowed, one might take 4.

Reply:
Quite possibly the solution here is actually the same as in the next comment below:
> The number we are looking in this case is probably simply the number of binary operators set to non-zero frequency.

Maybe the current choice between 4 and 2 here is from a time when `BackImpl` didn't yet exist, so there were either 4 or 2 binary operators. And when `BackImpl` was introduced, it was missed to change the 4 to 5.